include/HPDDM_schwarz.hpp

```diff
@@ -1137,6 +1136,58 @@ class Schwarz : public Preconditioner<
                     PetscCall(PetscObjectReference((PetscObject)empty));
                     PetscCall(STSetKSP(st, ksp[0]));
                 }
+                if(levels[0]->threshold >= 0.0) {
+                    flg = PETSC_FALSE;
+                    PetscCall(PetscOptionsGetBool(nullptr, prefix, "-eps_use_inertia", &flg, nullptr));
+                    if(flg) {
+                        KSP           ksp;
+                        PC            pc;
+                        Mat           F;
+                        MatSolverType type;
+                        PetscInt      val;
+                        PetscCall(EPSSetTarget(eps, levels[0]->threshold));
+                        PetscCall(STGetOperator(st, nullptr));
+                        PetscCall(STGetKSP(st, &ksp));
+                        PetscCall(KSPGetPC(ksp, &pc));
+                        PetscCall(PCSetFromOptions(pc));
+                        PetscCall(PetscObjectTypeCompareAny((PetscObject)pc, &flg, PCLU, PCCHOLESKY, nullptr));
+                        PetscCheck(flg, PETSC_COMM_SELF, PETSC_ERR_SUP, "Cannot use -%seps_use_inertia without PCLU or PCCHOLESKY", prefix);
+                        if(PetscDefined(HAVE_MUMPS)) {
+                            PetscCall(PCFactorGetMatSolverType(pc, &type));
+                            if(!type) {
+                                PetscCall(PCFactorSetMatSolverType(pc, MATSOLVERMUMPS));
+                                PetscCall(PCSetFromOptions(pc));
+                            }
+                            else
+                                PetscCall(PetscStrcmp(type, MATSOLVERMUMPS, &flg));
+                        }
+                        PetscCall(PCFactorGetMatrix(pc, &F));
```

Comment:
Shouldn't this line go after line 1177. Also, why don't you use `MatGetInertia` in the MUMPS case as well?

Reply:
Indeed, maybe I get the order wrong, I will double check. I'm not using the higher-level API in the MUMPS case for two reasons:
1. it looks a little bit fragile to me, e.g., it will error out with a `PCLU` while the number of off-diagonal pivots may still be a good approximation for `-eps_nev`;
2. theoretically (does not really apply here because this is currently called on communicators of size 1 so ScaLAPACK is never used), I have to enforce `ICNTL(13) = 1` (that makes me realize there is an error in the code I will fix, should be `MatMumpsSetIcntl(F, 13, 1)`), so the MUMPS-specific branch has to be there anyway.

Reply:
Both the call order and the ICNTL value are now fixed.